xds/src/test/java/io/grpc/xds/EnvoyServerProtoDataTest.java

```diff
@@ -93,6 +95,37 @@ public void listener_convertFromListenerProto() {
     assertThat(tlsCertSdsConfigs.get(0).getName()).isEqualTo("google-sds-config-default");
   }
 
+  // TODO(sanjaypujare): remove when we move to envoy protos v3
+  @Test
+  public void listener_convertFromDeprecatedListenerProto() throws InvalidProtocolBufferException {
+    io.envoyproxy.envoy.api.v2.core.Address address =
+        io.envoyproxy.envoy.api.v2.core.Address.newBuilder()
+            .setSocketAddress(SocketAddress.newBuilder()
+                .setPortValue(8000)
+                .setAddress("10.2.1.34")
+                .build())
+            .build();
+    io.envoyproxy.envoy.api.v2.Listener listener =
+        io.envoyproxy.envoy.api.v2.Listener.newBuilder()
+            .setName("8000")
+            .setAddress(address)
+            .addFilterChains(createDeprecatedInFilter())
+            .build();
+    Listener xdsListener = Listener.fromEnvoyProtoListener(listener);
+    List<EnvoyServerProtoData.FilterChain> filterChains = xdsListener.getFilterChains();
+    EnvoyServerProtoData.FilterChain inFilter = filterChains.get(0);
+    assertThat(inFilter).isNotNull();
```

Comment:
this always pass or it won't execute this line. i think you should check size of filterChains